compiler/rustc_typeck/src/coherence/orphan.rs

```diff
@@ -330,6 +344,24 @@ fn emit_orphan_check_error<'tcx>(
     })
 }
 
+fn emit_orphan_check_for_impl_on_types<'tcx, T: rustc_errors::EmissionGuarantee>(
```

Comment:
The name should suggest that this is raw ptr specific:

```suggestion
fn emit_newtype_suggestion_for_raw_ptr<'tcx, T: rustc_errors::EmissionGuarantee>(
```